core/src/main/java/hudson/model/Slave.java

```diff
@@ -348,9 +349,14 @@ public FilePath getRootPath() {
     /**
      * Web-bound object used to serve jar files for JNLP.
      */
-    public static final class JnlpJar implements HttpResponse {
+    public static class JnlpJar implements HttpResponse {
```

Comment:
It is for https://issues.jenkins-ci.org/browse/JENKINS-44101, see the TODO in the paired PR